third-party/patches/unibilium-Add-fallback-func.patch

```diff
@@ -0,0 +1,71 @@
+diff --git a/unibilium.h b/unibilium.h
+index 7d78fa1..67e704a 100644
+--- a/unibilium.h
++++ b/unibilium.h
+@@ -627,4 +627,7 @@ void unibi_format(
+ 
+ size_t unibi_run(const char *, unibi_var_t [9], char *, size_t);
+ 
++typedef unibi_term *Unibi_Fallback_Func(const char*);
++void unibi_set_fallback(Unibi_Fallback_Func *);
++
+ #endif /* GUARD_UNIBILIUM_H_ */
+diff --git a/uniutil.c b/uniutil.c
+index 2e266a7..e953787 100644
+--- a/uniutil.c
++++ b/uniutil.c
+@@ -39,6 +39,12 @@ enum {MAX_BUF = 4096};
+ 
+ const char *const unibi_terminfo_dirs = TERMINFO_DIRS;
+ 
++static Unibi_Fallback_Func *fallback_func = NULL;
++
++void unibi_set_fallback(Unibi_Fallback_Func *func) {
```

Comment:
Instead of a "fallback" hook, would it work to have something like : 

```c
void unibi_override(unibi_term *unibi) {
  override_unibi = unibi;
}
```

Then `unibi_from_term` simply skips itself if `override_unibi` is not NULL:

```c
unibi_term *unibi_from_term(const char *term) {
  if (override_unibi) {
    return override_unibi;
  }
  ...
}
```

Reply:
> `unibi_override`

Sounds like a good idea.

Reply:
@erw any comment ?

Reply:
I went through the code again and I don't think this is going to work. Because `unibi_destroy()` is called in [`load_terminfo()`](https://github.com/neovim/libtermkey/blob/b7fe3af141c53cce71a244282b69ea860452120a/driver-ti.c#L275) when calling [`termkey_start()`](https://github.com/neovim/libtermkey/blob/b7fe3af141c53cce71a244282b69ea860452120a/termkey.c#L492), `override_unibi` will be released. If we want to implement this, I think we need to deep copy `unibit_term`. As you can see from the beginning of [unibilium.c](https://github.com/neovim/unibilium/blob/0c5fa34f78503e23484da0294339ea1c209c4866/unibilium.c), deep-copying `unibi_term` seems a bit daunting.